### DISCUSSION TITLE
The size of tensor a (6300) must match the size of tensor b (6810)?

### DISCUSSION THREAD

**Discussion by StefanOltmann** [deepjavalibrary/djl#3012] [Q&A] [answered]
Using the RetinaFace sample I get this error message: 
`ai.djl.engine.EngineException: The size of tensor a (6300) must match the size of tensor b (6810) at non-singleton dimension 0`

What does it mean?

**Comment by zachgk** [comment, marked as answer]
It means that in some `NDArray` operation, you are using `NDArray`s of incompatible shapes. For example, you can't add an array of `Shape(3)` to one of `Shape(5)`. The shapes have to be equal or broadcastable to each other. The stack trace or more of the error message should give you some hints as to which operation is wrong

**Reply by StefanOltmann** [reply]
Thank you. I tracked the problem I had down and sent you an PR #3016. :)
